### PR TITLE
Adding processing option to classify 16bit rasters using scaled values instead of raw values.

### PR DESCRIPTION
**Pull request by sdlime**
Initial attempt at addressing issue #5830. Use would be something like:

```
    PROCESSING "SCALE=10,50"
    PROCESSING "SCALE_BUCKETS=256"
    PROCESSING "CLASSIFY_SCALED=TRUE"
```

Only supported with 16bit rasters.

--Steve

### CODE REVIEW

File: mapdrawgdal.c [hunk 1]
```diff
@@ -2141,6 +2156,17 @@ msDrawRasterLayerGDAL_16BitClassification(
             (starttime.tv_sec+starttime.tv_usec/1.0e6) );
   }
 
+  // testy...
+  if(1==1) {
```
Comment (rouault) [inline mapdrawgdal.c]: to remove

File: mapdrawgdal.c [hunk 2]
```diff
@@ -1998,6 +2001,13 @@ msDrawRasterLayerGDAL_16BitClassification(
     }
   }
 
+  /* -------------------------------------------------------------------- */
+  /*      Fetch the scale classification option.                          */
+  /* -------------------------------------------------------------------- */
+  pszClassifyScaled = CSLFetchNameValue( layer->processing, "CLASSIFY_SCALED" );
+  if( pszClassifyScaled != NULL && EQUAL(pszClassifyScaled,"TRUE") )
```
Comment (rouault) [inline mapdrawgdal.c]: Perhaps use
```
if( pszClassifyScaled != NULL )
   bClassifyScale = CSLTestBoolean(pszClassifyScaled);
```
so as to accept TRUE, YES, ON, 1

**Comment by jmckenna** [comment]
This looks ready to merge, any objections?  (to include in 7.6.0)

**Comment by sdlime** [comment]
Looks like I need to resolve @rouault comments. Will do so this evening.

File: mapdrawgdal.c (hunk 1 above)
Comment (sdlime) [reply mapdrawgdal.c]: Whoops, removed...

File: mapdrawgdal.c (hunk 2 above)
Comment (sdlime) [reply mapdrawgdal.c]: Ok, cool - a good pattern to know about.

**Comment by sdlime** [comment]
If things pass definitely will want to squash then merge...

**Comment by sdlime** [comment]
@jmckenna - this was merged.

**Comment by jmckenna** [comment]
thank you @sdlime !

**Pull request merged**